icu4c/source/i18n/unicode/decimfmt.h

```diff
@@ -2156,8 +2156,18 @@ class U_I18N_API DecimalFormat : public NumberFormat {
     //                                   INSTANCE FIELDS                                   //
     //=====================================================================================//
 
-    // Only one instance field: keep all fields inside of an implementation class defined in number_mapper.h
-    number::impl::DecimalFormatFields* fields;
+    /**
+    * An internal status variable used to indicate that the object is in an 'invalid' state.
+    * Required as there are several methods which internally allocate memory and don't have a 
+    * any error parameters.
+    * Also needs to be 'mutable' since there are 'const' methods with no error parameter
```

Comment:
Which const methods try allocating memory?  Is it just the parse methods?

Reply:
Right, its the `parse` and `parseCurrency` methods. Neither have an error code and both try to allocate memory. (This is the lazy-init `parser` member.)

Reply:
For that particular case, it seems to me that it would be simpler to make parse and parseCurrency just check to see if the lazy-init was successful and produced a non-null pointer.

If you do want a top-level internal error code field, maybe make it std::atomic?

Reply:
If parse or parseCurrency fail, it doesn't mean that the whole DecimalFormat needs to be in a failure state.  You can be happily formatting all you want and not have the parser working.

Reply:
I've force pushed changes and the `internalStatus` is now gone.